src/master-playlist-controller.js

```diff
@@ -446,6 +446,22 @@ export default class MasterPlaylistController extends videojs.EventTarget {
     this.audioPlaylistLoader_.start();
   }
 
+
+  /**
+   * Re-tune playback quality level for the current player
+   * conditions. This method may perform destructive actions, like
+   * removing already buffered content, to readjust the currently
+   * active playlist quickly.
+   */
+  fastQualityChange_() {
+    let media = this.hlsHandler.selectPlaylist();
+
+    if (media !== this.masterPlaylistLoader_.media()) {
+      this.masterPlaylistLoader_.media(media);
+      this.mainSegmentLoader_.sourceUpdater_.remove(this.currentTimeFunc() + 5, Infinity);
```

Comment:
I debated using the segment loader's bandwidth info and the bitrate of the newly selected bitrate to calculate a safe removal range. Seems like it might be a good idea but I thought it could wait for a separate PR. Seem reasonable?

Proxying remove() calls from the segment loader to the source updater increases encapsulation but I don't think it was worthwhile given all the actors are in the same project in this case.